src/main/java/com/google/devtools/build/lib/remote/disk/DiskCacheGarbageCollectorIdleTask.java

```diff
@@ -36,7 +36,7 @@ public final class DiskCacheGarbageCollectorIdleTask implements IdleTask {
   private final DiskCacheGarbageCollector gc;
 
   private static final ExecutorService executorService =
-      Executors.newCachedThreadPool(
+      Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors(),
```

Comment:
Since this workload is probably IO-bound, could you try how it performs if you use `newThreadPerTaskExecutor` instead (so that it uses virtual threads)?

@tjgq

Reply:
Doesn't the ForkJoinPool used by virtual threads default to the number of cpus? I think I'd prefer something like `newFixedThreadPool(min(4, availableProcessors()), ...)` so that we still get some amount of parallel I/O on single-cpu systems.

Reply:
Added performance info to CollectionStats: d1a835456d44ca3ce9e9d2944a2eb2e68e1ce75e

This is the performance for on a MacBook Pro M1 Max for the different options:
1. `newFixedThreadPool(Runtime.getRuntime().availableProcessors(), ...`:

```
241027 16:43:49.855:I 471 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Disk cache garbage collection started
241027 16:45:02.959:I 471 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Deleted 505571 of 505571 files, reclaimed 19.0 GiB of 19.0 GiB in 13.09 seconds (38626 files/s, 1484 MB/s)
```
2. unbounded virtual threads: `Executors.newThreadPerTaskExecutor(Thread.ofVirtual()...`:
```
241027 18:16:24.806:I 472 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Disk cache garbage collection started
241027 18:17:38.658:I 472 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Deleted 505572 of 505572 files, reclaimed 19.0 GiB of 19.0 GiB in 13.83 seconds (36546 files/s, 1404 MB/s)
```

and this the performance on a linux machine where the thread error was not happening:

0. original `newCachedThreadPool`:
```
241027 18:10:55.309:I 569 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Disk cache garbage collection started
241027 18:10:59.136:I 569 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Deleted 505556 of 505556 files, reclaimed 18.9 GiB of 18.9 GiB in 3.81 seconds (132622 files/s, 5090 MB/s)
```
1. `newFixedThreadPool(Runtime.getRuntime().availableProcessors(), ...`:
```
241027 19:54:50.313:I 561 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Disk cache garbage collection started
241027 19:54:53.935:I 561 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Deleted 505555 of 505555 files, reclaimed 18.9 GiB of 18.9 GiB in 3.61 seconds (140121 files/s, 5378 MB/s)
```
2. unbounded virtual threads: `Executors.newThreadPerTaskExecutor(Thread.ofVirtual()...`:
```
241027 18:54:57.419:I 560 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Disk cache garbage collection started
241027 18:55:01.136:I 560 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Deleted 505556 of 505556 files, reclaimed 18.9 GiB of 18.9 GiB in 3.70 seconds (136637 files/s, 5244 MB/s)
```

Ok to use the `Executors.newThreadPerTaskExecutor(Thread.ofVirtual()...` option?: d9afaa0f24f7891aa94a61fff9b51f662d54bc00

Reply:
I mean, the numbers suggest that virtual threads have slightly worse performance, so considering that this will have to be cherry-picked into 7.4.1, let's be conservative and stick with non-virtual threads for now?

Also, please do use `min(4, cpus)` as the pool size; in the past I've seen very bad performance on similar filesystem-bound thread pools when running on single or dual core machines.